experiment/gerrit-onboarder/onboarder.go

```diff
@@ -0,0 +1,357 @@
+	orderedKeys = append(orderedKeys, id)
+	return mapToGroups(groupsMap, orderedKeys)
+}
+
+func updateGroups(uuid, group string) (bool, error) {
+	data, err := ioutil.ReadFile(GROUPS_FILE)
+	if err != nil {
+		return false, fmt.Errorf("faield to read groups file: %w", err)
+	}
+
+	newData := ensureUUID(string(data), uuid, group)
+	err = ioutil.WriteFile(GROUPS_FILE, []byte(newData), 0755)
+	if err != nil {
+		return false, fmt.Errorf("faield to write groups file: %w", err)
+	}
+
+	return newData != string(data), nil
+}
+
+func configToMap(configFile string) (map[string][]string, []string) {
+	configMap := map[string][]string{}
+	orderedKeys := []string{}
+	var curKey string
+	if configFile == "" {
+		return configMap, orderedKeys
+	}
+	for _, line := range strings.Split(configFile, "\n") {
+		if strings.HasPrefix(line, "[") {
+			curKey = line
+			orderedKeys = append(orderedKeys, line)
+		} else {
+			if curList, ok := configMap[curKey]; ok {
+				configMap[curKey] = append(curList, line)
+			} else {
+				configMap[curKey] = []string{line}
+			}
+		}
+	}
+	return configMap, orderedKeys
+}
+
+func mapToConfig(configMap map[string][]string, orderedIDs []string) string {
+	res := ""
+	for _, header := range orderedIDs {
+		res = res + header + "\n"
+		for _, line := range configMap[header] {
+			res = res + line + "\n"
+		}
+	}
+	if res == "" {
+		return res
+	}
+	return strings.TrimSpace(res) + "\n"
+}
+
+func contains(s []string, v string) bool {
+	for _, item := range s {
+		if strings.TrimSpace(item) == strings.TrimSpace(v) {
+			return true
+		}
+	}
+	return false
+}
+
+func getInheritedRepo(configMap map[string][]string) string {
+	if section, ok := configMap["[access]"]; ok {
+		for _, line := range section {
+			if strings.Contains(line, "inheritFrom") {
+				return strings.TrimSpace(strings.Split(line, "=")[1])
+			}
+		}
+	}
+	return ""
+}
+
+func addSection(header string, configMap map[string][]string, configOrder, neededLines []string) (map[string][]string, []string) {
+	if _, ok := configMap[header]; !ok {
+		configMap[header] = []string{}
+		configOrder = append(configOrder, header)
+	}
+	for _, line := range neededLines {
+		configMap[header] = append(configMap[header], "\t"+line)
+	}
+
+	return configMap, configOrder
+}
+
+func labelExists(configMap map[string][]string) bool {
+	_, ok := configMap[LABEL_HEADER]
+	return ok
+}
+
+func lineInRightHeaderFunc(header, line string) func(map[string][]string) bool {
+	return func(configMap map[string][]string) bool {
+		if value, ok := configMap[header]; ok && contains(value, line) {
+			return true
+		}
+		return false
+	}
+}
+
+func labelAccessExistsFunc(groupName string) func(map[string][]string) bool {
+	return func(configMap map[string][]string) bool {
+		for _, value := range configMap {
+			for _, item := range value {
+				if strings.HasPrefix(strings.TrimSpace(item), "label-Verified =") && strings.HasSuffix(strings.TrimSpace(item), fmt.Sprintf("group %s", groupName)) {
```

Comment:
A comment here would be helpful to readers, briefly explaining that this tool is best effort and not to be intrusive. So when target label is already associated with prow robot, then assume it's already done, and no PR is needed.

Also, please use either a global var or an option for the name of the label

Reply:
Done